### ISSUE TITLE
opt: invalid SPIR-V produced: The continue construct with the continue target 32[%32] is not post dominated by the back-edge block 51[%51]

### ISSUE THREAD

**Issue by paulthomson** [KhronosGroup/SPIRV-Tools#2711]
[bug_report.zip](https://github.com/KhronosGroup/SPIRV-Tools/files/3355619/bug_report.zip)

Issue found using [GraphicsFuzz](https://github.com/google/graphicsfuzz).

Tool versions:

* glslangValidator commit hash: 40c16ec0b3ad03fc170f1369a58e7bbe662d82cd
* spirv-opt commit hash: 4a00a80c40484a6f6f72f48c9d34943cf8f180d4

To reproduce:

`glslangValidator -V shader.frag -o shader.frag.spv`

`spirv-opt shader.frag.spv -o temp.spv --validate-after-all --eliminate-local-single-block --reduce-load-size --combine-access-chains --eliminate-local-multi-store --reduce-load-size --convert-local-access-chains --eliminate-dead-branches --eliminate-dead-branches --merge-return --copy-propagate-arrays --copy-propagate-arrays --eliminate-dead-inserts --eliminate-dead-branches --merge-blocks --copy-propagate-arrays --combine-access-chains --simplify-instructions --convert-local-access-chains --inline-entry-points-exhaustive`

The following shader files are included in the attached archive, some of which are also shown inline below:

* 0_glsl/shader.frag
* 1_spirv/shader.frag.spv
* 1_spirv_asm/shader.frag.asm

0_glsl/shader.frag:

```
#version 310 es

precision highp float;

vec3 mand()
{
 for(
     int k = 0;
     k < 1000;
     1
 )
  {
   discard;
  }
 return vec3(1.0);
}
void main()
{
 for(
     int j = 0;
     j < 4;
     1
 )
  {
   mand();
  }
}

```

**Comment by s-perron** [comment]
I need to test, but I believe this is fixed by https://github.com/KhronosGroup/SPIRV-Tools/pull/2713.

**Comment by s-perron** [comment]
Just tried it and it seems to be passing.

**Comment by paulthomson** [comment]
Re-opening since the fix was reverted, and otherwise I might report this issue again.

**Comment by s-perron** [comment]
Fixed by #2842.
